### PR TITLE
test: use node helpers for events.once and setTimeout promise

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change

Node has these helpers, we might as well replace our bespoke ones with these built-in ones!

Notes: none

### CODE REVIEW

File: spec/lib/spec-helpers.ts [hunk 1]
```diff
@@ -21,8 +21,6 @@ const addOnly = <T>(fn: Function): T => {
 export const ifit = (condition: boolean) => (condition ? it : addOnly<TestFunction>(it.skip));
 export const ifdescribe = (condition: boolean) => (condition ? describe : addOnly<SuiteFunction>(describe.skip));
 
-export const delay = (time: number = 0) => new Promise(resolve => setTimeout(resolve, time));
```
Comment (codebytere) [inline spec/lib/spec-helpers.ts:24]:
```diff
diff --git a/spec/lib/spec-helpers.ts b/spec/lib/spec-helpers.ts
index 485f960f58..6c4d1a774a 100644
--- a/spec/lib/spec-helpers.ts
+++ b/spec/lib/spec-helpers.ts
@@ -181,6 +181,7 @@ export async function itremote (name: string, fn: Function, args?:
 any[]) {
     const { ok, message } = await w.webContents.executeJavaScript(`(async () => {
       try {
         const chai_1 = require('chai')
+        const promises_1 = require('timers/promises')
         chai_1.use(require('chai-as-promised'))
         chai_1.use(require('dirty-chai'))
         await (${fn})(...${JSON.stringify(args ?? [])})
```

fixes the webview spec failure

Comment (nornagon) [reply spec/lib/spec-helpers.ts:24]: wanna make that a suggestion on the right line in this PR? :)

Comment (codebytere) [reply spec/lib/spec-helpers.ts:24]: I can't - you can't make suggestions on lines of a given PR that aren't changed

Comment (nornagon) [reply spec/lib/spec-helpers.ts:24]: oh d'oh 🤦🏻‍♂️ thanks!

Comment (codebytere) [reply spec/lib/spec-helpers.ts:24]: i would have if i could have 😅

**Comment by codebytere** [comment]
@nornagon looks like the crash reporter tests are failing on a similar error:

Screenshot 2023-02-23 at 9 59 10 AM


Adding `const promises_1 = require('timers/promises')` to the relevant files seems to fix it - for ex for `respects an overridden crash dump directory`:

```diff
diff --git a/spec/fixtures/apps/remote-control/main.js b/spec/fixtures/apps/remote-control/main.js
index b8d89ff724..59a65b4fa8 100644
--- a/spec/fixtures/apps/remote-control/main.js
+++ b/spec/fixtures/apps/remote-control/main.js
@@ -1,6 +1,7 @@
 const { app } = require('electron');
 const http = require('http');
 const v8 = require('v8');
+const promises_1 = require('timers/promises')
 
 if (app.commandLine.hasSwitch('boot-eval')) {
   // eslint-disable-next-line no-eval

```

**Review by codebytere — APPROVED** [review]

**Pull request merged**